eclass/ecm-common.eclass

```diff
@@ -0,0 +1,347 @@
+# via ecm_install_icons(ICONS <icon> DESTINATION <icon_install_dir)
+if [[ ${ECM_INSTALL_ICONS} ]]; then
+	[[ ${ECM_INSTALL_ICONS@a} == *a* ]] ||
+		die "ECM_INSTALL_ICONS must be an array"
+fi
+
+# @ECLASS_VARIABLE: ECM_KCM_TARGETS
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Array of <target>:<subdir> tuples to feed to ECMInstallIcons via
+# ecmcommon_generate_desktop_file(<target> <subdir>), which is this
+# eclass adaptation of kcmutils_generate_desktop_file.
+if [[ ${ECM_KCM_TARGETS} ]]; then
+	[[ ${ECM_KCM_TARGETS@a} == *a* ]] ||
+		die "ECM_KCM_TARGETS must be an array"
+fi
+
+DESCRIPTION="Common files for ${PN/-common/}"
+
+BDEPEND=">=kde-frameworks/extra-cmake-modules-5.115.0:*"
+
+case ${ECM_I18N} in
+	true)
+		KF5_BDEPEND+=( "kde-frameworks/ki18n:5" )
+		KF6_BDEPEND+=( "kde-frameworks/ki18n:6" )
+		;;
+	false) ;;
+	*)
+		eerror "Unknown value for \${ECM_I18N}"
+		die "Value ${ECM_I18N} is not supported"
+		;;
+esac
+
+case ${ECM_HANDBOOK} in
+	true)
+		IUSE+=" +handbook"
+		KF5_BDEPEND+=( "handbook? ( kde-frameworks/kdoctools:5 )" )
+		KF6_BDEPEND+=( "handbook? ( kde-frameworks/kdoctools:6 )" )
+		;;
+	false) ;;
+	*)
+		eerror "Unknown value for \${ECM_HANDBOOK}"
+		die "Value ${ECM_HANDBOOK} is not supported"
+		;;
+esac
+
+if [[ ${ECM_KCM_TARGETS} ]]; then
+	KF5_BDEPEND+=( "kde-frameworks/kcmutils:5" )
+	KF6_BDEPEND+=( "kde-frameworks/kcmutils:6" )
+fi
+
+if [[ ${KF6_BDEPEND} && ${KF5_BDEPEND} ]]; then
+	BDEPEND+=" || ( ( ${KF6_BDEPEND[*]} ) ( ${KF5_BDEPEND[*]} ) )"
+fi
+
+# @FUNCTION: _ecm-common_preamble
+# @INTERNAL
+# @DESCRIPTION:
+# Create a CMakeLists.txt file with minimum ECM setup.
+_ecm-common_preamble() {
+	cat > CMakeLists.txt <<- _EOF_ || die
+		cmake_minimum_required(VERSION 3.16)
+		project(${PN} VERSION ${PV})
+
+		find_package(ECM "5.115.0" REQUIRED NO_MODULE)
+		set(CMAKE_MODULE_PATH \${ECM_MODULE_PATH})
+
+		# Set by pkg_setup(); Use this if need to differ between KF5 or KF6
+		set(KFSLOT ${_KFSLOT})
+		set(KDE_INSTALL_DOCBUNDLEDIR "${EPREFIX}/usr/share/help" CACHE PATH "")
+
+		include(KDEInstallDirs)
+		include(ECMOptionalAddSubdirectory) # commonly used
+		include(FeatureSummary)
+	_EOF_
+
+	if [[ ${ECM_INSTALL_ICONS} ]]; then
+		cat >> CMakeLists.txt <<- _EOF_ || die
+			include(ECMInstallIcons)
+		_EOF_
+	fi
+}
+
+# @FUNCTION: _ecm-common_i18n
+# @INTERNAL
+# @DESCRIPTION:
+# Find KF[56]I18n and call ki18n_install(po).
+_ecm-common_i18n() {
+	[[ ${ECM_I18N} == true ]] || return
+	cat >> CMakeLists.txt <<- _EOF_ || die
+		find_package(KF\${KFSLOT}I18n REQUIRED)
+		ki18n_install(po)
+	_EOF_
+}
+
+# @FUNCTION: _ecm-common_docs
+# @INTERNAL
+# @DESCRIPTION:
+# Find KF[56]DocTools, call kdoctools_install(po) and
+# add_subdirectory(${ECM_HANDBOOK_DIRS})
+_ecm-common_docs() {
+	( in_iuse handbook && use handbook ) || return
+	local i
+
+	cat >> CMakeLists.txt <<- _EOF_ || die
+		find_package(KF\${KFSLOT}DocTools REQUIRED)
+		kdoctools_install(po)
+	_EOF_
+
+	if [[ ${ECM_HANDBOOK_DIRS} ]]; then
```

Comment:
I don't think we need this condition. We could early-return if it's false - which would look nicer - but the nested for/if should be enough anyway.